tests/test_nubank_client.py

```diff
@@ -683,19 +683,56 @@ def test_get_qr_code():
     assert uid != ''
     assert isinstance(qr, QRCode)
 
-
-@pytest.mark.parametrize("http_status", [
+not_accepted_http_codes = [
     100, 101, 102, 103,
     201, 202, 203, 204, 205, 206, 207, 208, 226,
     300, 301, 302, 303, 304, 305, 306, 307, 308,
     400, 401, 402, 403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 413, 414, 415, 416, 417, 418, 420, 421, 422,
     423,
     424, 426, 428, 429, 431, 440, 444, 449, 450, 451, 495, 496, 497, 498, 499,
     500, 501, 502, 503, 504, 505, 506, 507, 508, 509, 510, 511, 520, 521, 522, 523, 524, 525, 526, 527, 530, 598
-])
+]
+
+@pytest.mark.parametrize("http_status", not_accepted_http_codes)
 @patch.object(Nubank, '_update_proxy_urls', fake_update_proxy)
 def test_nubank_request_handler_throws_exception_on_status_different_of_200(http_status):
     response = create_fake_response({}, http_status)
     client = Nubank()
     with pytest.raises(NuException):
         client._handle_response(response)
+
+@pytest.mark.parametrize("http_status", not_accepted_http_codes)
```

Comment:
Do we need to test all http codes ? We already have a test that check if all codes throw the exception, maybe just one here should be fine

Reply:
I might be overdoing but I did this to avoid hard coding an error code into the message/NuException.status_code.
If you want I can make it only for code 400

Reply:
Just one status code is fine, since the mechanism is already tested by the `test_nubank_request_handler_throws_exception_on_status_different_of_200`.

Another option would be merge this test with the previous one and instead of just check that an exception is thrown, also check the status code contained in it.